libpromises/cf3lex.l

```diff
@@ -252,11 +258,11 @@ promise_type   [a-zA-Z_]+:
                                    return 0;
                                  }
                                  P.if_depth--;
-                               }
-
-<if_ignore_state>[^@]      {
-                               ParserDebug("\tL:inside macro @if, ignoring text:%s\n", yytext);
-                           }
+                               }\
+<if_ignore_state>.    {
+                          /* eat up al unknown chars when line starts with @*/
+                          ParserDebug("\tL:inside macro @if, ignoring text:%s\n", yytext);
+                      }
```

Comment:
@tzz: You recently made a change to add this '[^@]'. Will it still work with Bas' change?